src/liballoc/vec.rs

```diff
@@ -603,6 +603,92 @@ impl<T> Vec<T> {
         self.buf.try_reserve_exact(self.len, additional)
     }
 
+    /// Allows re-interpreting the type of a Vec to reuse the allocation.
+    /// The vector is emptied and any values contained in it will be dropped.
+    /// The target type must have the same size and alignment as the source type.
+    /// This API doesn't transmute any values of T to U, because it makes sure
+    /// to empty the vector before any unsafe operations.
+    ///
+    /// # Example
+    ///
+    /// This API is useful especially when using `Vec` as a
+    /// temporary storage for data with short lifetimes.
+    /// By recycling the allocation, the `Vec` is able to safely
+    /// outlive the lifetime of the type that was stored in it.
+    /// ```
+    /// # #![feature(recycle_vec)]
+    /// # use std::error::Error;
+    /// #
+    /// # struct Stream(bool);
+    /// #
+    /// # impl Stream {
+    /// #     fn new() -> Self {
+    /// #         Stream(false)
+    /// #     }
+    /// #
+    /// #     fn next(&mut self) -> Option<&[u8]> {
+    /// #         if self.0 {
+    /// #             None
+    /// #         } else {
+    /// #             self.0 = true;
+    /// #             Some(&b"foo"[..])
+    /// #         }
+    /// #     }
+    /// # }
+    /// #
+    /// # fn process(input: &[Object<'_>]) -> Result<(), Box<dyn Error>> {
+    /// #     for obj in input {
+    /// #         let _ = obj.reference;
+    /// #     }
+    /// #     Ok(())
+    /// # }
+    /// #
+    /// # struct Object<'a> {
+    /// #     reference: &'a [u8],
+    /// # }
+    /// #
+    /// # fn deserialize<'a>(input: &'a [u8], output: &mut Vec<Object<'a>>) -> Result<(), Box<dyn Error>> {
+    /// #     output.push(Object { reference: input });
+    /// #     Ok(())
+    /// # }
+    /// #
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// # let mut stream = Stream::new();
+    /// let mut objects: Vec<Object<'static>> = Vec::new();    // Any lifetime goes here
+    ///
+    /// while let Some(byte_chunk) = stream.next() {           // `byte_chunk` lifetime starts
+    ///     let mut temp: Vec<Object<'_>> = objects.recycle(); // `temp` lifetime starts
+    ///
+    ///     // Zero-copy parsing; deserialized `Object`s have references to `byte_chunk`
```

Comment:
```suggestion
    ///     // Zero-copy parsing; deserialized `Object`s have references to `byte_chunk`.
```
Use complete sentences in standard library docs. :)